source/reading.Rmd

```diff
@@ -1159,124 +1135,176 @@ is that data owners have much more control over the data they provide to users.
 with how you write and run your code. You should also keep in mind that when a website owner
 grants you API access, they also usually specify a limit (or *quota*) of how much data you can ask for.
-Be careful not to overrun your quota! In this example, we should take a look at
- [the Twitter website](https://developer.twitter.com/en/docs/twitter-api/rate-limits) to see what limits
-we should abide by when using the API. 
-
-**Using `rtweet`**
-
-After checking the Twitter website, it seems like asking for 400 tweets one time is acceptable.
-So we can use the `get_timelines` function to ask for the last 400 tweets from the [\@tidyverse](https://twitter.com/tidyverse) account.
-
+Be careful not to overrun your quota! So *before* we try to use the API, we will first visit
+[the NASA website](https://api.nasa.gov/) to see what limits we should abide by when using the API.
+These limits are outlined in Figure \@ref(fig:NASA-API-limits).
+
+```{r NASA-API-limits, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "The NASA website specifies an hourly limit of 1,000 requests.", fig.retina = 2, out.width="100%"}
+knitr::include_graphics("img/reading/NASA-API-limits.png")
+```
+
+After checking the NASA website, it seems like we can send at most 1,000 requests per hour.
+That should be more than enough for our purposes in this section.
+
+#### Accessing the NASA API {-}
+
+The NASA API is what is known as an *HTTP API*: this is a particularly common
+kind of API, where you can obtain data simply by accessing a
+particular URL as if it were a regular website.  To make a query to the NASA
+API, we need to specify three things.  First, we specify the URL *endpoint* of
+the API, which is simply a URL that helps the remote server understand which
+API you are trying to access. NASA offers a variety of APIs, each with its own
+endpoint; in the case of the NASA "Astronomy Picture of the Day" API, the URL
+endpoint is `https://api.nasa.gov/planetary/apod`, as shown at the top of
+Figure \@ref(fig:NASA-API-parameters). Second, we write `?`, which denotes that a
+list of *query parameters* will follow. And finally, we specify a list of
+query parameters of the form `parameter=value`, separated by `&` characters. The NASA
+"Astronomy Picture of the Day" API accepts the parameters shown in
+Figure \@ref(fig:NASA-API-parameters). 
+
+```{r NASA-API-parameters, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = 'The set of parameters that you can specify when querying the NASA "Astronomy Picture of the Day" API, along with syntax, default settings, and a description of each.', fig.retina = 2, out.width="100%"}
+knitr::include_graphics("img/reading/NASA-API-parameters.png")
+```
+
+So for example, to obtain the image of the day
+from July 13, 2023, the API query would have two parameters: `api_key=YOUR_API_KEY`
+and `date=2023-07-13`.
+```
+https://api.nasa.gov/planetary/apod?api_key=YOUR_API_KEY&date=2023-07-13
```

Comment:
I know this is obvious, but maybe we should add a reminder for the students to replace YOUR_API_KEY with their API key. Something like "Remember to replace `YOUR_API_KEY` with the API key you received in from NASA in your email!"